### PR TITLE
App/Service: Bound service to MainActivity 

### PR DESCRIPTION
**Pull request by niley7464**
This patch bounds service to MainActivity.

Reference: https://developer.android.com/develop/background-work/services/bound-services

### CODE REVIEW

File: ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainActivity.kt [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
-import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
-import ai.nnstreamer.ml.inference.offloading.ui.theme.NnstreamerandroidTheme
-import android.content.Intent
 
 class MainActivity : ComponentActivity() {
+    private val TAG = "MainActivity"
+    private lateinit var mService: MainService
+    private var mBound: Boolean = false
```
Comment (jaeyun-jung) [inline ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainActivity.kt:22]: Q: is it necessary?
if bound, mService in onServiceConnected() is not null.
when disconnected (onServiceDisconnected() or onStop()), mService = null;

then we can handle button event with condition mService is not null, and remove boolean flag mBound.

Comment (niley7464) [reply ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainActivity.kt]: Oh you're right.
I just followed the tutorial without thinking😅
`mBound` seems to be an unnecessary variable.
Thank you!

File: ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainActivity.kt [hunk 2]
```diff
@@ -25,7 +42,30 @@ class MainActivity : ComponentActivity() {
...
         startForegroundService(Intent(this, MainService::class.java))
+
+        val start = findViewById<Button>(R.id.start)
+        start.setOnClickListener(View.OnClickListener {
+            mService?.startServer()
+        })
+
+        val stop = findViewById<Button>(R.id.stop)
+        stop.setOnClickListener(View.OnClickListener {
+            mService?.startServer()
```
Comment (jaeyun-jung) [inline ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainActivity.kt:56]: it would be `stopServer()` :)

Comment (niley7464) [reply ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainActivity.kt]: Sorry.
I fixed it

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**